### PR TITLE
Feature/126 install docs

### PR DESCRIPTION
**Pull request by JohnHalleyGotway**
## Expected Differences ##

- [x] Do these changes modify the system output in any way? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
I spent a long time testing this behavior in the METplus AWS account. WRF Cloud is current installed/running at app.metpluscloud.org.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

@hahnd: Please primarily review the non-documentation edits in `install_bootstrap.sh` and note that `cf_imagebuilder_wrf_intel.yaml` is probably NOT working as expected. Please advise how to correctly define 	`${GIT_CLONE_OPTS}` in that conext.

@bruckerd and @fossell, please review this updated [System Installation](https://wrfcloud--136.org.readthedocs.build/en/136/Users_Guide/installation.html) chapter of the User's Guide. Please feel free to recommend edits, re-wording, or additional details, as needed.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**
No automated testing, but manual testing at app.metpluscloud.org.

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[Wed 2/15/23]**.</br>

## Pull Request Checklist ##
- [x] Review the source issue metadata (labels, project, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Project**
Select: **Milestone** as the version that will include these changes
Select: **Development** to link to the original development issue.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Users_Guide/installation.rst [hunk 1]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+
+5. Copy and paste the following into the CloudShell terminal:
+
+  .. code-block:: ini
+
+    git clone --branch ${GITHUB_NAME} https://github.com/NCAR/wrfcloud
+    ./wrfcloud/install_bootstrap.sh ${GITHUB_NAME}
+
+This bootstrap script takes about 25 minutes to run. It is followed by a series of interactive questions that must be completed prior to WRF Cloud being installed. For each question, the default option (if applicable) is provided in sqaure braces. Simply hit enter to accept the default or modify the setting as needed. These questions include:
+
```
Comment (fossell) [inline docs/Users_Guide/installation.rst:101]: Should we add info about what these things are? E.g. what does enabling autocompletion mean? Where should this info exist, i.e. here within each question or elsewhere?

File: docs/Users_Guide/installation.rst [hunk 2]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+  * Would you like to enable autocompletion? [*Recommend Yes*]
+  * Which domain name would you like to use? [`Route 53 <https://aws.amazon.com/route53>`_ domain(s) from :numref:`prerequisites`]
+  * Enter host name for web application: [e.g. app.{DOMAIN}]
```
Comment (fossell) [inline docs/Users_Guide/installation.rst:104]: Should we somehow indicate that {DOMAIN} is what was entered in the previous question or from prereqs?

File: docs/Users_Guide/installation.rst [hunk 3]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+  * Enter host name for REST API: [e.g. api.{DOMAIN}]
+  * Enter host name for websocket API: [e.g. ws.{DOMAIN}]
+  * Enter administrator's full name:
+  * Enter email address for application administrator:
+  * Enter administrator's new password:
+  * Do you want to install example model configurations? [*Recommend Yes*]
+  * Do you want to upload an SSH public key for an admin? [*Recommend Yes*]
+  * Paste your public key, often found at ${HOME}/.ssh/id_rsa.pub:
```
Comment (fossell) [inline docs/Users_Guide/installation.rst:112]: Add documentation link for creating one or is that overkill?

File: docs/Users_Guide/installation.rst [hunk 4]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+  * If directed to the **WRF Cloud Login** page, use the administrator email address and password defined above to login and proceed to the next step.
+  * If, however, your browser downloads a file named **download** rather than allowing you to login, follow the steps outlined below to fix this behavior.
+
+    a. In the CloudShell terminal, check that temporary build directory still exists. It should remain for a while after install but will eventually be scrubbed from **/tmp**.
+
+    .. code-block:: ini
+
+       ls /tmp/wrfcloud-build-*
+
+    b. If it exists, run the following commands to update the WRF Cloud web files:
```
Comment (fossell) [inline docs/Users_Guide/installation.rst:148]: Is there a condition where this doesn't exist but the user still experiences the weird download file behavior?

**Review by fossell — COMMENTED** [review]
@JohnHalleyGotway - Added a few comments/questions for clarification.

File: docs/Users_Guide/installation.rst (hunk 4 above)
Comment (JohnHalleyGotway) [reply docs/Users_Guide/installation.rst:148]: Yes, if they run the install steps, go home for the day, and check the next day (as I did), the temp files will have likely already been scrubbed.

Note that @hahnd is checking with AWS to see if this behavior can be avoided entirely. So ideally it'll be resolved another way and we can remove this rather clunky patching step. But until we actually solve that, I think it's good to include these steps.

File: docs/Users_Guide/installation.rst (hunk 3 above)
Comment (JohnHalleyGotway) [reply docs/Users_Guide/installation.rst:112]: Do you want me to include a link to https://git-scm.com/book/en/v2/Git-on-the-Server-Generating-Your-SSH-Public-Key?

File: python/src/wrfcloud/aws/resources/cf_imagebuilder_wrf_intel.yaml [hunk 5]
```diff
@@ -314,7 +314,7 @@ Resources:
                     - source /etc/bashrc
                     - mkdir -p /opt/src
                     - cd /opt/src
-                    - git clone -b feature/112-model-configs https://github.com/NCAR/wrfcloud
+                    - git clone ${GIT_CLONE_OPTS} https://github.com/NCAR/wrfcloud
```
[the remark this answers is not in the document]
Comment (hahnd) [reply python/src/wrfcloud/aws/resources/cf_imagebuilder_wrf_intel.yaml:519]: Made these changes with commit ID: 945d9305f244636172d27c1704fe92ff7697fdae

**Review by hahnd — APPROVED** [review]

File: docs/Users_Guide/installation.rst [hunk 6]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+5. Copy and paste the following into the CloudShell terminal:
+
+  .. code-block:: ini
+
+    git clone --branch ${GITHUB_NAME} https://github.com/NCAR/wrfcloud
+    ./wrfcloud/install_bootstrap.sh ${GITHUB_NAME}
+
+This bootstrap script takes about 25 minutes to run. It is followed by a series of interactive questions that must be completed prior to WRF Cloud being installed. For each question, the default option (if applicable) is provided in square braces. Simply hit enter to accept the default or modify the setting as needed. These questions include:
+
+  * Would you like to enable autocompletion? [*Recommend Yes*]
```
Comment (fossell) [inline docs/Users_Guide/installation.rst:102]:
```suggestion
  * Would you like to enable autocompletion? [*Recommend Yes*]
     * This is required to install Angular, which creates the website. Please answer Yes.
```

**Review by fossell — CHANGES_REQUESTED** [review]
@JohnHalleyGotway - I provided some content suggestions.

File: docs/Users_Guide/installation.rst (hunk 1 above)
Comment (fossell) [reply docs/Users_Guide/installation.rst:101]: Added suggestions.

File: docs/Users_Guide/installation.rst [hunk 7]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+.. _prerequisites:
 
 Prerequisites
 =============
 
-* AWS Account
-* Domain name in Route 53
-* AWS account limit for "Running On-Demand All HPC instances" in us-east-2 must be 96 vCPUs or higher
-* Simple Email Service (SES) must be out of the sandbox in us-east-2
+The prerequisites for installing WRF Cloud in an AWS account are described below.
```
[the remark this answers is not in the document]
Comment (fossell) [reply docs/Users_Guide/installation.rst:35]: @hahnd - Can you please review the wording and suggest changes to make sure this is technically correct?

File: docs/Users_Guide/installation.rst [hunk 8]
```diff
@@ -7,35 +7,231 @@ System Installation
...
+  * Do you want to upload an SSH public key for an admin? [*Recommend Yes*]
+  * Paste your public key, often found at ${HOME}/.ssh/id_rsa.pub:
+
+    * Copy and paste the output of this command into the CloudShell terminal:
+
+      .. code-block:: ini
+
+        cat ${HOME}/.ssh/id_rsa.pub
+
+5. After completing these steps, check the email address provided above to verify it. It takes approximately 10-20 minutes for the website to become available.
```
Comment (fossell) [inline docs/Users_Guide/installation.rst:120]: The numbering is off, this is the second step 5.

**Comment by JohnHalleyGotway** [comment]
After incorporating recommend changes from @fossell and updates to this feature branch from @hahnd, I am proceeding with one final round of testing by following the PR version of the [System Installation](https://wrfcloud--136.org.readthedocs.build/en/136/Users_Guide/installation.html) instructions:

1. Uninstall the metpluscloud.org.
   - Tweaked the wording slightly.
   - Did have to redo deleting `WrfCloudApiData` because it couldn't be deleted until `WrfCloudWebApp` was gone.
   - Confirmed that I can no longer log into app.metpluscloud.org as expected.
2. Re-install at metpluscloud.org.
   - I note that the "autocompletion" question has disappeared. The first question is about the domain name. I'll remove the note about autocompletion.
   - The install went smooth and I was able to log on WITHOUT any odd download behavior. Recommend keeping the "download" behavior troubleshooting instructions until someone else confirms that they're no longer needed.
   - Confirmed that the sample carribean domain does now exist. So @hahnd's fix worked.
   - Confirmed that I can add new users (Kate, David, and George).
   - Found in the CloudFormation `Parameters` tab that `GitCloneOpts` was correctly set as `--branch feature/126-install-docs`.
   - Takes a long time for the EC2 image to build... still waiting.

**Comment by JohnHalleyGotway** [comment]
@fossell I did make some edits to the install docs since you last reviewed ([7d4284a](https://github.com/NCAR/wrfcloud/pull/136/commits/7d4284a77bafa00fd9d0170fe2a7492ca593d3d1), [6534979](https://github.com/NCAR/wrfcloud/pull/136/commits/6534979e0217f0ec81686909b92dde7beb8ca09d), [0c6138d](https://github.com/NCAR/wrfcloud/pull/136/commits/0c6138d90b725493502b470f4fb7ffaf19b17e4f)). But I think we're in good shape now.

**Review by fossell — APPROVED** [review]
Updates to install guide look good. I'm also able to activate an account for this install test, indicating the install was successful from that standpoint.

**Pull request merged**